<h1 align="center">Atliq Hotels Data Analysis Project<h1>

In [126]:
import pandas as pd

## lets read all the data

In [127]:
dates_df = pd.read_csv("F:\\DataScience\\Git\\data\\dim_date.csv")
hotels_df = pd.read_csv("F:\\DataScience\\Git\\data\\dim_hotels.csv")
rooms_df = pd.read_csv("F:\\DataScience\\Git\\data\\dim_rooms.csv")
room_bookings_agg_df = pd.read_csv("F:\\DataScience\\Git\\data\\fact_aggregated_bookings.csv")
bookings_df = pd.read_csv("F:\\DataScience\\Git\\data\\fact_bookings.csv")
property_occupancy_df = pd.read_csv("F:\\DataScience\\Git\\data\\new_data_august.csv")

***
### ==> 1. Data Import and Data Exploration
***

In [128]:
dates_df.head()

date  mmm yy week no  day_type
0  01-May-22  May 22    W 19   weekend
1  02-May-22  May 22    W 19  weekeday
2  03-May-22  May 22    W 19  weekeday
3  04-May-22  May 22    W 19  weekeday
4  05-May-22  May 22    W 19  weekeday

In [129]:
hotels_df.head()

property_id  property_name  category    city
0        16558   Atliq Grands    Luxury   Delhi
1        16559  Atliq Exotica    Luxury  Mumbai
2        16560     Atliq City  Business   Delhi
3        16561      Atliq Blu    Luxury   Delhi
4        16562      Atliq Bay    Luxury   Delhi

In [130]:
rooms_df.head()

room_id    room_class
0     RT1      Standard
1     RT2         Elite
2     RT3       Premium
3     RT4  Presidential

In [131]:
room_bookings_agg_df.head()

property_id check_in_date room_category  successful_bookings  capacity
0        16559      1-May-22           RT1                   25      30.0
1        19562      1-May-22           RT1                   28      30.0
2        19563      1-May-22           RT1                   23      30.0
3        17558      1-May-22           RT1                   30      19.0
4        16558      1-May-22           RT1                   18      19.0

In [132]:
bookings_df.head()

booking_id  property_id booking_date check_in_date checkout_date  \
0  May012216558RT11        16558     27-04-22      1/5/2022      2/5/2022   
1  May012216558RT12        16558     30-04-22      1/5/2022      2/5/2022   
2  May012216558RT13        16558     28-04-22      1/5/2022      4/5/2022   
3  May012216558RT14        16558     28-04-22      1/5/2022      2/5/2022   
4  May012216558RT15        16558     27-04-22      1/5/2022      2/5/2022   

   no_guests room_category booking_platform  ratings_given booking_status  \
0       -3.0           RT1    direct online            1.0    Checked Out   
1        2.0           RT1           others            NaN      Cancelled   
2        2.0           RT1          logtrip            5.0    Checked Out   
3       -2.0           RT1           others            NaN      Cancelled   
4        4.0           RT1    direct online            5.0    Checked Out   

   revenue_generated  revenue_realized  
0              10010             10010  
1               9100              3640  
2            9100000              9100  
3               9100              3640  
4              10920             10920

In [133]:
property_occupancy_df.head()

property_id  property_name  category       city room_category room_class  \
0        16559  Atliq Exotica    Luxury     Mumbai           RT1   Standard   
1        19562      Atliq Bay    Luxury  Bangalore           RT1   Standard   
2        19563   Atliq Palace  Business  Bangalore           RT1   Standard   
3        19558   Atliq Grands    Luxury  Bangalore           RT1   Standard   
4        19560     Atliq City  Business  Bangalore           RT1   Standard   

  check_in_date  mmm yy week no  day_type  successful_bookings  capacity  \
0     01-Aug-22  Aug-22    W 32  weekeday                   30        30   
1     01-Aug-22  Aug-22    W 32  weekeday                   21        30   
2     01-Aug-22  Aug-22    W 32  weekeday                   23        30   
3     01-Aug-22  Aug-22    W 32  weekeday                   30        40   
4     01-Aug-22  Aug-22    W 32  weekeday                   20        26   

     occ%  
0  100.00  
1   70.00  
2   76.67  
3   75.00  
4   76.92

In [134]:
print(dates_df.shape)
print(hotels_df.shape)
print(rooms_df.shape)
print(room_bookings_agg_df.shape)
print(bookings_df.shape)
print(property_occupancy_df.shape)

(92, 4)
(25, 4)
(4, 2)
(9200, 5)
(134590, 12)
(7, 13)


In [135]:
bookings_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 134590 entries, 0 to 134589
Data columns (total 12 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   booking_id         134590 non-null  object 
 1   property_id        134590 non-null  int64  
 2   booking_date       134590 non-null  object 
 3   check_in_date      134590 non-null  object 
 4   checkout_date      134590 non-null  object 
 5   no_guests          134587 non-null  float64
 6   room_category      134590 non-null  object 
 7   booking_platform   134590 non-null  object 
 8   ratings_given      56683 non-null   float64
 9   booking_status     134590 non-null  object 
 10  revenue_generated  134590 non-null  int64  
 11  revenue_realized   134590 non-null  int64  
dtypes: float64(2), int64(3), object(7)
memory usage: 12.3+ MB


In [136]:
bookings_df.describe()

property_id      no_guests  ratings_given  revenue_generated  \
count  134590.000000  134587.000000   56683.000000       1.345900e+05   
mean    18061.113493       2.036170       3.619004       1.537805e+04   
std      1093.055847       1.034885       1.235009       9.303604e+04   
min     16558.000000     -17.000000       1.000000       6.500000e+03   
25%     17558.000000       1.000000       3.000000       9.900000e+03   
50%     17564.000000       2.000000       4.000000       1.350000e+04   
75%     18563.000000       2.000000       5.000000       1.800000e+04   
max     19563.000000       6.000000       5.000000       2.856000e+07   

       revenue_realized  
count     134590.000000  
mean       12696.123256  
std         6928.108124  
min         2600.000000  
25%         7600.000000  
50%        11700.000000  
75%        15300.000000  
max        45220.000000

In [137]:
bookings_df.columns

Index(['booking_id', 'property_id', 'booking_date', 'check_in_date',
       'checkout_date', 'no_guests', 'room_category', 'booking_platform',
       'ratings_given', 'booking_status', 'revenue_generated',
       'revenue_realized'],
      dtype='object')

In [138]:
pd.Series(bookings_df.property_id.unique())

0     16558
1     16559
2     16560
3     16561
4     16562
5     16563
6     17558
7     17559
8     17560
9     17561
10    17562
11    17563
12    18558
13    18559
14    18560
15    18561
16    18562
17    18563
18    19558
19    19559
20    19560
21    19561
22    19562
23    19563
24    17564
dtype: int64

In [139]:
bookings_df.value_counts("property_id").sort_values(ascending=False).head()

property_id
16559    7338
18562    7333
16563    7147
18560    6638
18561    6458
Name: count, dtype: int64


### Dataset Descriptions

- **dates_df:** Contains date-related information (e.g., date keys, calendar dates, day, month, year, etc.).
- **hotels_df:** Hotel details, including hotel name, hotel ID, category (e.g., luxury, budget), and location (city or region).
- **rooms_df:** Room information, such as room ID and room class/type (e.g., Standard, Elite, Suite).
- **room_bookings_agg_df:** Aggregated booking data by room, including room ID, check-in date, room class, number of bookings, and room capacity.
- **bookings_df:** Individual booking records, with booking ID, property (hotel) ID, booking date, check-in and check-out dates, number of guests, room class/category, booking platform, customer ratings, booking status, revenue generated, and revenue realized.
- **property_occupancy_df:** Enhanced property-level data, including property ID, hotel name, category, city, room class/category, check-in date, month, year, week number, day type (e.g., weekend/weekday), number of successful bookings, total capacity, and occupancy percentage.


***
### ==> 2. Data Cleaning
***

In [140]:
dates_df.info() # convert date column to datetime
dates_df['date'] = pd.to_datetime(dates_df['date'], format='%d-%b-%y')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92 entries, 0 to 91
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   date      92 non-null     object
 1   mmm yy    92 non-null     object
 2   week no   92 non-null     object
 3   day_type  92 non-null     object
dtypes: object(4)
memory usage: 3.0+ KB


In [141]:
dates_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92 entries, 0 to 91
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   date      92 non-null     datetime64[ns]
 1   mmm yy    92 non-null     object        
 2   week no   92 non-null     object        
 3   day_type  92 non-null     object        
dtypes: datetime64[ns](1), object(3)
memory usage: 3.0+ KB


In [142]:
dates_df.head()

date  mmm yy week no  day_type
0 2022-05-01  May 22    W 19   weekend
1 2022-05-02  May 22    W 19  weekeday
2 2022-05-03  May 22    W 19  weekeday
3 2022-05-04  May 22    W 19  weekeday
4 2022-05-05  May 22    W 19  weekeday

In [143]:
room_bookings_agg_df["check_in_date"].head()

0    1-May-22
1    1-May-22
2    1-May-22
3    1-May-22
4    1-May-22
Name: check_in_date, dtype: object

In [144]:
room_bookings_agg_df['check_in_date'] = pd.to_datetime(room_bookings_agg_df['check_in_date'], format='%d-%b-%y')

In [145]:
room_bookings_agg_df["check_in_date"].head()

0   2022-05-01
1   2022-05-01
2   2022-05-01
3   2022-05-01
4   2022-05-01
Name: check_in_date, dtype: datetime64[ns]

In [146]:
bookings_df.head()

booking_id  property_id booking_date check_in_date checkout_date  \
0  May012216558RT11        16558     27-04-22      1/5/2022      2/5/2022   
1  May012216558RT12        16558     30-04-22      1/5/2022      2/5/2022   
2  May012216558RT13        16558     28-04-22      1/5/2022      4/5/2022   
3  May012216558RT14        16558     28-04-22      1/5/2022      2/5/2022   
4  May012216558RT15        16558     27-04-22      1/5/2022      2/5/2022   

   no_guests room_category booking_platform  ratings_given booking_status  \
0       -3.0           RT1    direct online            1.0    Checked Out   
1        2.0           RT1           others            NaN      Cancelled   
2        2.0           RT1          logtrip            5.0    Checked Out   
3       -2.0           RT1           others            NaN      Cancelled   
4        4.0           RT1    direct online            5.0    Checked Out   

   revenue_generated  revenue_realized  
0              10010             10010  
1               9100              3640  
2            9100000              9100  
3               9100              3640  
4              10920             10920

In [147]:
bookings_df['check_in_date'] = pd.to_datetime(bookings_df['check_in_date'], dayfirst=True, errors='coerce')
bookings_df['checkout_date'] = pd.to_datetime(bookings_df['checkout_date'], dayfirst=True, errors='coerce')
bookings_df['booking_date'] = pd.to_datetime(bookings_df['booking_date'], dayfirst=True, errors='coerce')

C:\Users\Lenoo\AppData\Local\Temp\ipykernel_12792\2715515514.py:3: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  bookings_df['booking_date'] = pd.to_datetime(bookings_df['booking_date'], dayfirst=True, errors='coerce')


In [148]:
bookings_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 134590 entries, 0 to 134589
Data columns (total 12 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   booking_id         134590 non-null  object        
 1   property_id        134590 non-null  int64         
 2   booking_date       134590 non-null  datetime64[ns]
 3   check_in_date      55804 non-null   datetime64[ns]
 4   checkout_date      54651 non-null   datetime64[ns]
 5   no_guests          134587 non-null  float64       
 6   room_category      134590 non-null  object        
 7   booking_platform   134590 non-null  object        
 8   ratings_given      56683 non-null   float64       
 9   booking_status     134590 non-null  object        
 10  revenue_generated  134590 non-null  int64         
 11  revenue_realized   134590 non-null  int64         
dtypes: datetime64[ns](3), float64(2), int64(3), object(4)
memory usage: 12.3+ MB


In [149]:
bookings_df["no_guests"].value_counts()

no_guests
 2.0     60219
 1.0     44311
 3.0     15119
 4.0     12094
 6.0      1444
 5.0      1391
-4.0         2
-3.0         1
-2.0         1
-10.0        1
-12.0        1
-6.0         1
-17.0        1
-1.0         1
Name: count, dtype: int64

In [150]:
bookings_df['no_guests'] = bookings_df['no_guests'].abs()

In [151]:
bookings_df["no_guests"].value_counts()

no_guests
2.0     60220
1.0     44312
3.0     15120
4.0     12096
6.0      1445
5.0      1391
10.0        1
12.0        1
17.0        1
Name: count, dtype: int64

In [152]:
bookings_df["booking_platform"].value_counts()

booking_platform
others            55066
makeyourtrip      26898
logtrip           14756
direct online     13379
tripster           9630
journey            8106
direct offline     6755
Name: count, dtype: int64

i found some outliers in revenue generated cloumn

In [153]:
bookings_df['revenue_generated'].max()

28560000

In [154]:
#using std deviation method
mean = bookings_df['revenue_generated'].mean()
std = bookings_df['revenue_generated'].std()
high = mean + 3 * std
low = mean - 3 * std
#bookings_df=bookings_df[(bookings_df['revenue_generated'] > low) & (bookings_df['revenue_generated']<high)]
bookings_df['revenue_generated']= bookings_df['revenue_generated'].clip(lower=low, upper=high).round(2)

In [155]:
bookings_df.revenue_generated.max()

294486.17

In [156]:
bookings_df["revenue_realized"].max()

45220

In [157]:
#using std deviation method
mean = bookings_df['revenue_realized'].mean()
std = bookings_df['revenue_realized'].std()
high = mean + 3 * std
low = mean - 3 * std
bookings_df=bookings_df[(bookings_df['revenue_realized'] > low) & (bookings_df['revenue_realized']<high)]

In [158]:
bookings_df["revenue_realized"].max()

32300

In [159]:
property_occupancy_df.head()

property_id  property_name  category       city room_category room_class  \
0        16559  Atliq Exotica    Luxury     Mumbai           RT1   Standard   
1        19562      Atliq Bay    Luxury  Bangalore           RT1   Standard   
2        19563   Atliq Palace  Business  Bangalore           RT1   Standard   
3        19558   Atliq Grands    Luxury  Bangalore           RT1   Standard   
4        19560     Atliq City  Business  Bangalore           RT1   Standard   

  check_in_date  mmm yy week no  day_type  successful_bookings  capacity  \
0     01-Aug-22  Aug-22    W 32  weekeday                   30        30   
1     01-Aug-22  Aug-22    W 32  weekeday                   21        30   
2     01-Aug-22  Aug-22    W 32  weekeday                   23        30   
3     01-Aug-22  Aug-22    W 32  weekeday                   30        40   
4     01-Aug-22  Aug-22    W 32  weekeday                   20        26   

     occ%  
0  100.00  
1   70.00  
2   76.67  
3   75.00  
4   76.92

In [160]:
property_occupancy_df.check_in_date=pd.to_datetime(property_occupancy_df.check_in_date, format='%d-%b-%y')

In [161]:
property_occupancy_df.head()

property_id  property_name  category       city room_category room_class  \
0        16559  Atliq Exotica    Luxury     Mumbai           RT1   Standard   
1        19562      Atliq Bay    Luxury  Bangalore           RT1   Standard   
2        19563   Atliq Palace  Business  Bangalore           RT1   Standard   
3        19558   Atliq Grands    Luxury  Bangalore           RT1   Standard   
4        19560     Atliq City  Business  Bangalore           RT1   Standard   

  check_in_date  mmm yy week no  day_type  successful_bookings  capacity  \
0    2022-08-01  Aug-22    W 32  weekeday                   30        30   
1    2022-08-01  Aug-22    W 32  weekeday                   21        30   
2    2022-08-01  Aug-22    W 32  weekeday                   23        30   
3    2022-08-01  Aug-22    W 32  weekeday                   30        40   
4    2022-08-01  Aug-22    W 32  weekeday                   20        26   

     occ%  
0  100.00  
1   70.00  
2   76.67  
3   75.00  
4   76.92

***
### ==> 3. Data Transformation
***

**Create occupancy percentage column**

In [162]:
room_bookings_agg_df.head()

property_id check_in_date room_category  successful_bookings  capacity
0        16559    2022-05-01           RT1                   25      30.0
1        19562    2022-05-01           RT1                   28      30.0
2        19563    2022-05-01           RT1                   23      30.0
3        17558    2022-05-01           RT1                   30      19.0
4        16558    2022-05-01           RT1                   18      19.0

In [163]:
room_bookings_agg_df['occ_percent'] = room_bookings_agg_df.apply(
    lambda row: (row['successful_bookings'] / row['capacity']) * 100, axis=1
).round(1)

In [164]:
room_bookings_agg_df.head()

property_id check_in_date room_category  successful_bookings  capacity  \
0        16559    2022-05-01           RT1                   25      30.0   
1        19562    2022-05-01           RT1                   28      30.0   
2        19563    2022-05-01           RT1                   23      30.0   
3        17558    2022-05-01           RT1                   30      19.0   
4        16558    2022-05-01           RT1                   18      19.0   

   occ_percent  
0         83.3  
1         93.3  
2         76.7  
3        157.9  
4         94.7

In [165]:
room_bookings_agg_df.shape

(9200, 6)

In [166]:
property_occupancy_df = property_occupancy_df.rename(columns={'occ%': 'occ_percent'})

In [167]:
property_occupancy_df.head()

property_id  property_name  category       city room_category room_class  \
0        16559  Atliq Exotica    Luxury     Mumbai           RT1   Standard   
1        19562      Atliq Bay    Luxury  Bangalore           RT1   Standard   
2        19563   Atliq Palace  Business  Bangalore           RT1   Standard   
3        19558   Atliq Grands    Luxury  Bangalore           RT1   Standard   
4        19560     Atliq City  Business  Bangalore           RT1   Standard   

  check_in_date  mmm yy week no  day_type  successful_bookings  capacity  \
0    2022-08-01  Aug-22    W 32  weekeday                   30        30   
1    2022-08-01  Aug-22    W 32  weekeday                   21        30   
2    2022-08-01  Aug-22    W 32  weekeday                   23        30   
3    2022-08-01  Aug-22    W 32  weekeday                   30        40   
4    2022-08-01  Aug-22    W 32  weekeday                   20        26   

   occ_percent  
0       100.00  
1        70.00  
2        76.67  
3        75.00  
4        76.92

In [168]:
merged_df = pd.merge(
    property_occupancy_df,
    room_bookings_agg_df,
    on=['property_id', 'check_in_date', 'room_category'],
    how='outer',  # use 'inner' if you want only matching rows
    suffixes=('_property', '_room')
)

In [169]:
merged_df.head()

property_id property_name category city room_category room_class  \
0        16558           NaN      NaN  NaN           RT1        NaN   
1        16558           NaN      NaN  NaN           RT2        NaN   
2        16558           NaN      NaN  NaN           RT3        NaN   
3        16558           NaN      NaN  NaN           RT4        NaN   
4        16558           NaN      NaN  NaN           RT1        NaN   

  check_in_date mmm yy week no day_type  successful_bookings_property  \
0    2022-05-01    NaN     NaN      NaN                           NaN   
1    2022-05-01    NaN     NaN      NaN                           NaN   
2    2022-05-01    NaN     NaN      NaN                           NaN   
3    2022-05-01    NaN     NaN      NaN                           NaN   
4    2022-05-02    NaN     NaN      NaN                           NaN   

   capacity_property  occ_percent_property  successful_bookings_room  \
0                NaN                   NaN                      18.0   
1                NaN                   NaN                      21.0   
2                NaN                   NaN                       8.0   
3                NaN                   NaN                       3.0   
4                NaN                   NaN                      12.0   

   capacity_room  occ_percent_room  
0           19.0              94.7  
1           22.0              95.5  
2            8.0             100.0  
3            3.0             100.0  
4           19.0              63.2

In [170]:
bookings_df.head()

booking_id  property_id booking_date check_in_date checkout_date  \
0  May012216558RT11        16558   2022-04-27    2022-05-01    2022-05-02   
1  May012216558RT12        16558   2022-04-30    2022-05-01    2022-05-02   
2  May012216558RT13        16558   2022-04-28    2022-05-01    2022-05-04   
3  May012216558RT14        16558   2022-04-28    2022-05-01    2022-05-02   
4  May012216558RT15        16558   2022-04-27    2022-05-01    2022-05-02   

   no_guests room_category booking_platform  ratings_given booking_status  \
0        3.0           RT1    direct online            1.0    Checked Out   
1        2.0           RT1           others            NaN      Cancelled   
2        2.0           RT1          logtrip            5.0    Checked Out   
3        2.0           RT1           others            NaN      Cancelled   
4        4.0           RT1    direct online            5.0    Checked Out   

   revenue_generated  revenue_realized  
0           10010.00             10010  
1            9100.00              3640  
2          294486.17              9100  
3            9100.00              3640  
4           10920.00             10920

In [171]:
bookings_df['real_revenue'] = bookings_df['revenue_generated'] - bookings_df['revenue_realized']

In [172]:
bookings_df.head()

booking_id  property_id booking_date check_in_date checkout_date  \
0  May012216558RT11        16558   2022-04-27    2022-05-01    2022-05-02   
1  May012216558RT12        16558   2022-04-30    2022-05-01    2022-05-02   
2  May012216558RT13        16558   2022-04-28    2022-05-01    2022-05-04   
3  May012216558RT14        16558   2022-04-28    2022-05-01    2022-05-02   
4  May012216558RT15        16558   2022-04-27    2022-05-01    2022-05-02   

   no_guests room_category booking_platform  ratings_given booking_status  \
0        3.0           RT1    direct online            1.0    Checked Out   
1        2.0           RT1           others            NaN      Cancelled   
2        2.0           RT1          logtrip            5.0    Checked Out   
3        2.0           RT1           others            NaN      Cancelled   
4        4.0           RT1    direct online            5.0    Checked Out   

   revenue_generated  revenue_realized  real_revenue  
0           10010.00             10010          0.00  
1            9100.00              3640       5460.00  
2          294486.17              9100     285386.17  
3            9100.00              3640       5460.00  
4           10920.00             10920          0.00

***
### ==> 4. Insights Generation
***

### Total revenue Generated by the Hotels 

In [ ]:
bookings_df['real_revenue'].sum()
bookings_df['revenue_generated'].sum()

300173506.85

In [186]:
print(f" Total revenue : ₹ {bookings_df['revenue_realized'].sum()}")

 Total revenue : ₹ 1659595999


In [185]:
print(f" Total Loss or Gap : ₹ {bookings_df['real_revenue'].sum()}")

 Total Loss or Gap : ₹ 300173506.85


In [181]:
profit_percentage = (bookings_df['revenue_realized'].sum() / bookings_df['revenue_generated'].sum()) * 100
print(f"Profit Percentage: {profit_percentage:.2f}%")

Profit Percentage: 84.68%


In [188]:
loss_percentage = (bookings_df['real_revenue'].sum() / bookings_df['revenue_generated'].sum()) * 100
print(f" Loss Percentage: {loss_percentage:.2f}%")

 Loss Percentage: 15.32%


### Total revenue : ₹ 1659595999
### Total Loss or Gap : ₹ 300173506.85
### Profit Percentage: 84.68%
### Loss Percentage: 15.32%

## Average revenue by room_category

In [207]:
# Calculate average real revenue by room_category
avg_revenue_by_category = bookings_df.groupby('room_category')['real_revenue'].mean().reset_index(name='avg_real_revenue')

# Merge with rooms_df to get room_class
avg_revenue_with_class = avg_revenue_by_category.merge(rooms_df, left_on='room_category', right_on='room_id', how='left')

# Drop the redundant room_id column if you want
avg_revenue_with_class = avg_revenue_with_class.drop(columns=['room_id'])

pd.DataFrame(avg_revenue_with_class )

room_category  avg_real_revenue    room_class
0           RT1       1418.434228      Standard
1           RT2       2004.620308         Elite
2           RT3       2672.820198       Premium
3           RT4       4379.898057  Presidential

## Number of Bookings in all Platform

In [ ]:
result = bookings_df.groupby('booking_platform').size().reset_index(name='count').sort_values('count', ascending=False)
result

booking_platform  count
5           others  54527
4     makeyourtrip  26633
3          logtrip  14619
1    direct online  13253
6         tripster   9534
2          journey   8035
0   direct offline   6689

## 2. Average occupancy rate in each of the room categories?

In [ ]:
avg_occ = room_bookings_agg_df.groupby('room_category')['occ_percent'].mean().reset_index(name='avg_occ_percent')

# Merge with rooms_df to get room_class
avg_occ_with_class = avg_occ.merge(rooms_df, left_on='room_category', right_on='room_id', how='left')

# Drop the redundant room_id column if you want
avg_occ_with_class = avg_occ_with_class.drop(columns=['room_id'])

avg_occ_with_class

room_category  avg_occ_percent    room_class
0           RT1        58.224021      Standard
1           RT2        58.040348         Elite
2           RT3        58.028783       Premium
3           RT4        59.303609  Presidential

In [250]:
merged_dfs= bookings_df.merge(hotels_df ,left_on ='property_id' ,right_on='property_id',how='left')
merged_dfs.head()

booking_id  property_id booking_date check_in_date checkout_date  \
0  May012216558RT11        16558   2022-04-27    2022-05-01    2022-05-02   
1  May012216558RT12        16558   2022-04-30    2022-05-01    2022-05-02   
2  May012216558RT13        16558   2022-04-28    2022-05-01    2022-05-04   
3  May012216558RT14        16558   2022-04-28    2022-05-01    2022-05-02   
4  May012216558RT15        16558   2022-04-27    2022-05-01    2022-05-02   

   no_guests room_category booking_platform  ratings_given booking_status  \
0       3.00           RT1    direct online           1.00    Checked Out   
1       2.00           RT1           others            NaN      Cancelled   
2       2.00           RT1          logtrip           5.00    Checked Out   
3       2.00           RT1           others            NaN      Cancelled   
4       4.00           RT1    direct online           5.00    Checked Out   

   revenue_generated  revenue_realized  real_revenue property_name category  \
0           10010.00             10010          0.00  Atliq Grands   Luxury   
1            9100.00              3640       5460.00  Atliq Grands   Luxury   
2          294486.17              9100     285386.17  Atliq Grands   Luxury   
3            9100.00              3640       5460.00  Atliq Grands   Luxury   
4           10920.00             10920          0.00  Atliq Grands   Luxury   

    city  
0  Delhi  
1  Delhi  
2  Delhi  
3  Delhi  
4  Delhi

## Total Revenue generated by Hotels in each city

In [252]:
pd.set_option('display.float_format', '{:.2f}'.format)
merged_dfs.groupby('city')['real_revenue'].sum().reset_index(name='Total Revenue INR').sort_values('Total Revenue INR', ascending=False).round(2)

city  Total Revenue INR
3     Mumbai       116773721.34
0  Bangalore        74431125.00
2  Hyderabad        56167980.00
1      Delhi        52800680.51

In [253]:
pd.set_option('display.float_format', '{:.2f}'.format)
merged_dfs.groupby('property_name')['real_revenue'].sum().reset_index(name='Total Revenue INR').sort_values('Total Revenue INR', ascending=False).round(2)

property_name  Total Revenue INR
3  Atliq Exotica        54948119.34
5   Atliq Palace        53814642.00
2     Atliq City        50835216.00
1      Atliq Blu        45782523.00
0      Atliq Bay        45746799.34
4   Atliq Grands        37506437.17
6  Atliq Seasons        11539770.00

## Room Booked vs Cancelled Rate

In [261]:
s= merged_dfs.groupby('booking_status')['booking_status'].count().reset_index(name='Total Bookings').sort_values('Total Bookings', ascending=False)
s['percent']=(s['Total Bookings']/s['Total Bookings'].sum())*100
s['percent'] = s['percent'].round(2)
s

booking_status  Total Bookings  percent
1    Checked Out           93186    69.91
0      Cancelled           33420    25.07
2        No Show            6684     5.01

# Atliq Hotels Data Analysis Project <br>
### This project analyzes booking and revenue data for Atliq Hotels, providing actionable business insights. <br>

**Key Insights** <br>
**Total Revenue:** ₹1,659,595,999 <br>
**Profit Percentage:** 84.68% <br>
**Loss Percentage (Cancellations/No-shows):** 15.32% <br>
**Highest Revenue City:** Mumbai <br>
**Top Performing Property:** Atliq Exotica <br>
**Highest Revenue Room Class:** Presidential <br>
**Average Occupancy Rate:** ~58-59% across all room types <br>
**Booking Status:** 25% cancellations, 5% no-shows <br>
**Most Bookings via:** Third-party platforms <br>
## Recommendations: <br>
Reduce cancellations and no-shows with better policies and guest engagement. <br>
Promote direct bookings to reduce commission costs. <br>
Expand premium room offerings and focus on high-performing cities/properties. <br>
Address underperforming properties with targeted strategies. <br>

# Thankssssssssss Broooooooooo